crates/languages/src/rust.rs

```diff
@@ -426,6 +437,23 @@ impl ContextProvider for RustContextProvider {
                 tags: vec!["rust-mod-test".to_owned()],
                 ..TaskTemplate::default()
             },
+            TaskTemplate {
+                label: format!(
+                    "cargo run -p {} --bin {}",
+                    RUST_PACKAGE_TASK_VARIABLE.template_value(),
+                    RUST_BIN_NAME_TASK_VARIABLE.template_value(),
+                ),
+                command: "cargo".into(),
+                args: vec![
+                    "run".into(),
+                    "-p".into(),
+                    RUST_PACKAGE_TASK_VARIABLE.template_value(),
+                    "--bin".into(),
+                    RUST_BIN_NAME_TASK_VARIABLE.template_value(),
```

Comment:
Do we need `RUST_BIN_NAME_TASK_VARIABLE` even? I think you could just use `ZED_STEM`; `cargo metadata` can take a while, and context evaluation should be as quick as possible.

Reply:
Yes, cargo run does not support specifying a bin using a path. That's why it's necessary to retrieve the bin name from the metadata.

```shell
❯ cargo run --help

Usage: cargo run [OPTIONS] [ARGS]...

Target Selection:
      --bin [<NAME>]      Name of the bin target to run
```

Reply:
`ZED_STEM` is not the full path though; it's just the name of the file, without a `.rs` extension. AFAIK there's no way to have a bin with a name different than it's stem, so IMHO it'd be better to use stem instead of invoking `cargo metadata`.

Reply:
You can do [e.g. this](https://github.com/jansol/texpresso/blob/de748cd341a8e14ea0a89070469e09588ca720bd/cli/Cargo.toml#L14), if I understood the question correctly?

Reply:
Hi @osiewicz @jansol 

Currently, the discussion is about the need to specify the bin target configured in the `Cargo.toml` when running `cargo run`.

For example, with an additional bin configuration in `Cargo.toml`:

```toml
[[bin]]
name = "bin-name"
path = "src/bin.rs"
```

To run the main function in `src/bin.rs`, we need to use `cargo run --bin bin-name`, otherwise, an error will occur.

![截屏2024-06-18 08 20 24](https://github.com/zed-industries/zed/assets/21101490/8d3b10a8-4430-479c-84a4-9da002440ba1)

Reply:
Alright, thanks you both for clearing that up and I'm sorry; I was wrong. 
I wonder though if we could avoid having to run `cargo metadata` at all when there's no main function in the current function; as is, we'll do it for each time we create task context for Rust.